crates/bevy_render/src/batching/mod.rs

```diff
@@ -69,58 +68,31 @@ pub fn batch_and_prepare_render_phase<I: CachedRenderPipelinePhaseItem, F: GetBa
 ) {
     let gpu_array_buffer = gpu_array_buffer.into_inner();
 
-    let mut process_item = |item: &mut I| -> Option<BatchMeta<F::CompareData>> {
-        let Ok((no_batching, batch_data)) = query.get(item.entity()) else {
-            return None;
-        };
-
-        let (user_data, buffer_data) = F::get_batch_data(batch_data);
+    let mut process_item = |item: &mut I| {
+        let (no_auto_batching, batch_query_item) = query.get(item.entity()).ok()?;
```

Comment:
It's probably important to call `F::get_batch_data`, even when the entity has a `NoAutomaticBatching`. I was confused by the naming here, why call `get_batch_data` when `no_batching` is true?… Well because it's not really no batching, but no *automatic* batching.

Reply:
We could call it prepare_and_get_batch_data perhaps. It happens to involve pushing the transform into the GpuBufferArray as implemented. Technically that isn't a requirement of getting the batch meta though, rather a convenient place to do it.

Reply:
maybe we should have separate functions for the CompareData and the BufferData? the compare data is generated even when we already know we won't be batching. would also be a bit clearer why we still need to call the `get_buffer` function

Reply:
Split into two functions.